src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -303,12 +211,12 @@ mp::QemuVirtualMachine::~QemuVirtualMachine()
 {
     update_shutdown_status = false;
     if (state == State::running)
+    {
         suspend();
-    else
-        shutdown();
+        vm_process->wait_for_finished();
+    }
 
     remove_tap_device(QString::fromStdString(tap_device_name));
-    vm_process->wait_for_finished();
```

Comment:
So now Process's destruction deals with reaping the qemu process?

Reply:
@Saviq,

Not sure what you mean here.  I found the former logic to be incorrect, irregardless of this PR, albeit this PR exposed it.  The new logic is basically saying to reap the qemu process only when its running.

Reply:
@townsend2010 I mean that we're not calling `shutdown()` any more, so is it that `Process`/`QProcess`'s d'tor handling killing the process?

Reply:
@Saviq,

This is what I mean by the incorrect logic.  If the process is running, we want to `suspend()` the instance when the dtor is called, ie, the daemon is exiting.  Otherwise, there is nothing to do and the process is not running.  The `shutdown()` there was really unnecessary the begin with.

Reply:
@townsend2010 and we don't care about any of the other states the vm might be in?

Reply:
@Saviq, hmm, ok I see your point.  For other "running" types of states, I suppose it should shut down the instance.  However, I'm not sure for `delayed_shutdown`. Do we suspend and when the daemon starts again, the instance is just `running`?  Or do we just shut it down and not start on daemon start?

Reply:
Ok, I added the logic back in, but wrapped it in a check to make sure `vm_process` is not null.